src/coreclr/jit/lowerarmarch.cpp

```diff
@@ -1303,17 +1303,32 @@ GenTree* Lowering::LowerHWIntrinsic(GenTreeHWIntrinsic* node)
         {
             GenTree* user = use.User();
             // Wrap the intrinsic in ConditionalSelect only if it is not already inside another ConditionalSelect
-            if (!user->OperIsHWIntrinsic() || (user->AsHWIntrinsic()->GetHWIntrinsicId() != NI_Sve_ConditionalSelect))
+            // If it is inside ConditionalSelect, then make sure that it is the `mask` operation of it.
```

Comment:
Not sure I follow this entirely...

This is stating we need to wrap in a `CndSel` if the user isn't a `CndSel` or if it is but we're not the mask operand.

The `!user->OperIsHWIntrinsic(NI_Sve_ConditionalSelect)` seems fine with that regard but the `HWIntrinsicInfo::ReturnsPerElementMask(node->GetHWIntrinsicId())` doesn't seem to fit.

Most notably we can have `mask = CndSel(mask1, mask2, mask3)` or `vector CndSel(mask, vector1, vector2)`, but also I would have expected that intrinsics like `Abs(x)` which must be emitted with a mask to need `CndSel` regardless of where it appears and only that we could optimize it away in some cases like `vector = CndSel(mask, vector1, CndSel(mask, vector2, vector3))`  (where the outer and inner cndsel use identical masks that mean the latter CndSel becomes redundant as the computed operands would never be selected anyways).

Reply:
So I would have expected the latter condition to basically be conditioned differently, basically ensuring that intrinsics that require a CndSel wrapper always, optionally trying to fold in the case the mask usages can be detected as redundant

Reply:
It's better to rewrite the comment to avoid confusion I guess:

```c++
          // Wrap the intrinsic in ConditionalSelect only if it is not already inside another ConditionalSelect
          // or if it is but it is the `mask` operand.
```

The goal of this block of code is to wrap `HWIntrinsic`s which satisfy `HWIntrinsicInfo::IsEmbeddedMaskedOperation(intrinsicId))` into `ConditionalSelect`. Usually this only needs to be done when the intrinsic is not already wrapped. But this logic doesn't hold when such intrinsic is used as the `mask` operand for `ConditionalSelect` - `user->AsHWIntrinsic()->GetHWIntrinsicId() != NI_Sve_ConditionalSelect` doesn't hold true any more, but the intrinsic still has to be wrapped `ConditionalSelect` (all that satisfy `HWIntrinsicInfo::IsEmbeddedMaskedOperation(intrinsicId)` have to).